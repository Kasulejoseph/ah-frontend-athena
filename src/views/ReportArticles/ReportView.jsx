import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { MDBBtn, MDBInput } from "mdbreact";
import { ReportArticleAction } from "../../actions/ReportArticleAction";

export class ReportView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      reason: ""
    };
  }

  handleSubmit = event => {
    event.preventDefault();
    const { reason } = this.state;
    const { ReportArticleAction, article } = this.props;
    ReportArticleAction(article.slug, reason);
  };

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {

    const { reason } = this.state;
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <div className="grey-text">
            <MDBInput
              name="reason"
              label="Enter reason here"
              group
              type="text"
              validate
              error="wrong"
              success="right"
              required
              value={reason}
              onChange={this.handleChange}
            />
          </div>
          <MDBBtn className="btn" color="primary" type="submit">
            Submit
          </MDBBtn>
        </form>
      </div>
    );
  }
}

ReportView.propTypes = {
  ReportArticleAction: PropTypes.func.isRequired,
  article: PropTypes.shape({}).isRequired
}


export default connect(
  null,
  {
    ReportArticleAction
  }
)(ReportView);
