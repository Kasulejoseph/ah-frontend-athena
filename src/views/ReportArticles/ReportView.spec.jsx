import React from "react";
import { shallow } from "enzyme";
import { ReportView } from "./ReportView";

let props = {
  ReportArticleAction: jest.fn(),
  article: {}
};
let wrapper;

describe("test report article view", () => {
  beforeEach(() => {
    wrapper = shallow(<ReportView {...props} />);
  });
  it("should match snapshots", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should mock handle submit", () => {
    wrapper.instance().handleSubmit({ preventDefault() {} });
    expect(props.ReportArticleAction).toBeCalled()
  });
  it('should mock handle change', () => {
    wrapper.instance().handleChange({ target: { name: "reason", value: "bad article" } });
  })
});
