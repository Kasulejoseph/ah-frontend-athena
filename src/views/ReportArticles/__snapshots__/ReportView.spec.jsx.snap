// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test report article view should match snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportView
    ReportArticleAction={[MockFunction]}
    article={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <div
          className="grey-text"
        >
          <t
            className=""
            containerClass=""
            disabled={false}
            error="wrong"
            filled={false}
            gap={false}
            group={true}
            icon=""
            iconBrand={false}
            iconClass=""
            iconLight={false}
            iconRegular={false}
            label="Enter reason here"
            labelClass=""
            name="reason"
            onChange={[Function]}
            required={true}
            size=""
            success="right"
            tag="input"
            type="text"
            validate={true}
            value=""
            valueDefault=""
          />
        </div>
        <t
          className="btn"
          color="primary"
          tag="button"
          type="submit"
        >
          Submit
        </t>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="grey-text"
          >
            <t
              className=""
              containerClass=""
              disabled={false}
              error="wrong"
              filled={false}
              gap={false}
              group={true}
              icon=""
              iconBrand={false}
              iconClass=""
              iconLight={false}
              iconRegular={false}
              label="Enter reason here"
              labelClass=""
              name="reason"
              onChange={[Function]}
              required={true}
              size=""
              success="right"
              tag="input"
              type="text"
              validate={true}
              value=""
              valueDefault=""
            />
          </div>,
          <t
            className="btn"
            color="primary"
            tag="button"
            type="submit"
          >
            Submit
          </t>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <t
              className=""
              containerClass=""
              disabled={false}
              error="wrong"
              filled={false}
              gap={false}
              group={true}
              icon=""
              iconBrand={false}
              iconClass=""
              iconLight={false}
              iconRegular={false}
              label="Enter reason here"
              labelClass=""
              name="reason"
              onChange={[Function]}
              required={true}
              size=""
              success="right"
              tag="input"
              type="text"
              validate={true}
              value=""
              valueDefault=""
            />,
            "className": "grey-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "containerClass": "",
              "disabled": false,
              "error": "wrong",
              "filled": false,
              "gap": false,
              "group": true,
              "hint": undefined,
              "icon": "",
              "iconBrand": false,
              "iconClass": "",
              "iconLight": false,
              "iconRegular": false,
              "iconSize": undefined,
              "id": undefined,
              "label": "Enter reason here",
              "labelClass": "",
              "name": "reason",
              "onChange": [Function],
              "required": true,
              "size": "",
              "success": "right",
              "tag": "input",
              "type": "text",
              "validate": true,
              "value": "",
              "valueDefault": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Submit",
            "className": "btn",
            "color": "primary",
            "tag": "button",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <div
            className="grey-text"
          >
            <t
              className=""
              containerClass=""
              disabled={false}
              error="wrong"
              filled={false}
              gap={false}
              group={true}
              icon=""
              iconBrand={false}
              iconClass=""
              iconLight={false}
              iconRegular={false}
              label="Enter reason here"
              labelClass=""
              name="reason"
              onChange={[Function]}
              required={true}
              size=""
              success="right"
              tag="input"
              type="text"
              validate={true}
              value=""
              valueDefault=""
            />
          </div>
          <t
            className="btn"
            color="primary"
            tag="button"
            type="submit"
          >
            Submit
          </t>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="grey-text"
            >
              <t
                className=""
                containerClass=""
                disabled={false}
                error="wrong"
                filled={false}
                gap={false}
                group={true}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                label="Enter reason here"
                labelClass=""
                name="reason"
                onChange={[Function]}
                required={true}
                size=""
                success="right"
                tag="input"
                type="text"
                validate={true}
                value=""
                valueDefault=""
              />
            </div>,
            <t
              className="btn"
              color="primary"
              tag="button"
              type="submit"
            >
              Submit
            </t>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <t
                className=""
                containerClass=""
                disabled={false}
                error="wrong"
                filled={false}
                gap={false}
                group={true}
                icon=""
                iconBrand={false}
                iconClass=""
                iconLight={false}
                iconRegular={false}
                label="Enter reason here"
                labelClass=""
                name="reason"
                onChange={[Function]}
                required={true}
                size=""
                success="right"
                tag="input"
                type="text"
                validate={true}
                value=""
                valueDefault=""
              />,
              "className": "grey-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "containerClass": "",
                "disabled": false,
                "error": "wrong",
                "filled": false,
                "gap": false,
                "group": true,
                "hint": undefined,
                "icon": "",
                "iconBrand": false,
                "iconClass": "",
                "iconLight": false,
                "iconRegular": false,
                "iconSize": undefined,
                "id": undefined,
                "label": "Enter reason here",
                "labelClass": "",
                "name": "reason",
                "onChange": [Function],
                "required": true,
                "size": "",
                "success": "right",
                "tag": "input",
                "type": "text",
                "validate": true,
                "value": "",
                "valueDefault": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Submit",
              "className": "btn",
              "color": "primary",
              "tag": "button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
